### ISSUE TITLE
Dynamically rendering columns triggers multiple bugs

### ISSUE THREAD

**Issue by jelhan** [miguelcobain/ember-yeti-table#270]
I have a use case in which the user should be able to show and hide columns. For performance reasons I tried to add and remove the columns dynamically. By doing so I'm avoiding creating DOM for data, which is not shown at all. That's the main reason why I preferred that approach over toggling visibility via CSS.

I'm doing so by wrapping all components for that specific row in `{{if}}` blocks. A simplified example looks like the following:

```hbs
<YetiTable @data={{this.data}} as |table|>
  <table.header as |header|>
    <header.column @prop="name">
      Name
    </header.column>
     
    {{#if this.showCityColumn}}
      <header.column @prop="city">
        City
      </header.column>
    {{/if}}
  </table.header>

  <table.body/>
</YetiTable>
```

By doing so I noticed a couple of bugs:

1. If dynamically adding a `<header.column>` the related data is not shown in the table. It does not render the `<td>` elements at all.  
2. A filter used on a dynamically added row is not working.
3. If a row is added after the column has been shown (e.g. due to changing a filter value on another column) the column is rendered correctly for that row.
4. If manually rendering the table body, the data of the row is shown as expected but a filter for that row is not added.

All these bugs could be reproduced using this Ember Twiddle: https://ember-twiddle.com/2ac131c6549fab3962d475863a42b98c?openFiles=templates.application%5C.hbs%2C

**Comment by jelhan** [comment]
Just noticed that the `<header.column>` component has a `visible` argument. It seems to provide the functionality I'm looking for without triggering this bug. I haven't found it in the documentation. Only came across when reading the source code to debug. https://github.com/miguelcobain/ember-yeti-table/blob/d79cee141ddbacd8dfd1f8114d989edb53df0419/addon/components/yeti-table/thead/row/column/component.js#L65-L69

While debugging I noticed that `columns` property of `<YetiTable>` is not tracked. https://github.com/miguelcobain/ember-yeti-table/blob/d79cee141ddbacd8dfd1f8114d989edb53df0419/addon/components/yeti-table/component.js#L339-L342
But it's mutated when a new column is registered. https://github.com/miguelcobain/ember-yeti-table/blob/d79cee141ddbacd8dfd1f8114d989edb53df0419/addon/components/yeti-table/component.js#L602-L609
That might be the reason for the bug. But I have stopped debugging until I found the `visible` argument.

**Comment by miguelcobain** [comment]
There's a brief mention of it on the general section: https://miguelcobain.github.io/ember-yeti-table/docs/general

> Additionally, you might need to toggle the visibility of each row, and for that we can use the @visible argument on the <header.column> component. It defaults to true. Setting it to false will hide all the cells for that column accross all rows.

at some point the `@visible` argument was on the API docs of the [header.column component](https://miguelcobain.github.io/ember-yeti-table/docs/api/modules/components/yeti-table/thead/row/column/component~Column), but it seems like it is not picking it up anymore.

I'll keep this open so that I remember to add a failing test with your use case and hopefully come up with a fix.

**Comment by cah-briangantzler** [comment]
Im not sure I understand why you would want a failing test, the visible property does exactly what is request. I do not believe you should support conditional rendering of a column, as your snipped from the docs points out, they would have to ALSO condition the rendering of the column in the row, and if they have a footer row, condition the column there as well, this just leads to more code and possible problems. Thats why the visible property exists.

I think the solution is just to ensure the visible property is documented more completely

**Comment by jelhan** [comment]
@cah-briangantzler I agree that one might decide to not support such a conditional rendering. Using `visible` argument simplified my templates a lot. But in that case an assertion should be thrown if a developer tries to do so.
